### PR TITLE
Remove dead code

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description
This PR does the following:
-  Remove spring package
- Remove other dead code (e.g., websocket stuff)
- Move middleware.spark.Main to middleware.OtpMiddlewareMain
- Adds some swagger tests

It supersedes part of #37 (which is expected to take a bit longer).

### CODE REVIEW

File: src/test/java/org/opentripplanner/middleware/docs/SwaggerTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SwaggerTest extends OtpMiddlewareTest {
```
Comment (br648) [inline src/test/java/org/opentripplanner/middleware/docs/SwaggerTest.java]: Should this class/test be part of this PR? The test fails for me.

**Review by br648 — CHANGES_REQUESTED** [review]
I think these classes can be removed as well:

```
org.opentripplanner.middleware.BasicOtpDispatcher
org.opentripplanner.middleware.Const
org.opentripplanner.middleware.WebSocketClient
```

File: src/test/java/org/opentripplanner/middleware/docs/SwaggerTest.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/test/java/org/opentripplanner/middleware/docs/SwaggerTest.java]: Perfect segway fo me to updating swagger.

Comment (binh-dam-ibigroup) [reply src/test/java/org/opentripplanner/middleware/docs/SwaggerTest.java]: @landonreed for the purpose of this PR, we should remove this test (and bring it back to the portion of #37 concerning swagger).

**Comment by binh-dam-ibigroup** [comment]
@br648 please see #48 for the removed classes.

> I think these classes can be removed as well:
> 
> ```java
> org.opentripplanner.middleware.BasicOtpDispatcher
> org.opentripplanner.middleware.Const
> org.opentripplanner.middleware.WebSocketClient
> ```

**Review by binh-dam-ibigroup — APPROVED** [review]
Conditional approval upon merging #48.

File: src/test/java/org/opentripplanner/middleware/docs/SwaggerTest.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/test/java/org/opentripplanner/middleware/docs/SwaggerTest.java]: #48 removes this test file.

**Review by evansiroky — APPROVED** [review]

**Comment by binh-dam-ibigroup** [comment]
Oh one more thing before merging, please remove these two packages:
- spring (the PR description states it is removing the spring package, however it is still listed as a dependency in `pom.xml`...) 
- org.java-websocket.

Thanks!

**Review by br648 — APPROVED** [review]
All good.

**Pull request merged**